checkov/cloudformation/checks/resource/aws/KMSRotation.py

```diff
@@ -13,5 +13,16 @@ def __init__(self) -> None:
     def get_inspected_key(self) -> str:
         return "Properties/EnableKeyRotation"
 
+    def scan_resource_conf(self, conf):
+        # Only symmetric keys support auto rotation. The attribute is optional and defaults to symmetric.
+        if 'Properties' in conf.keys():
+            if 'KeySpec' in conf['Properties'].keys():
+                spec = conf['Properties']['KeySpec']
+                if not spec or 'SYMMETRIC_DEFAULT' in spec or 'HMAC' in spec:
+                    return super().scan_resource_conf(conf)
+                else:
+                    return CheckResult.PASSED
+        return super().scan_resource_conf(conf)
```

Comment:
```suggestion
        properties = conf.get("Properties")
        if properties and isinstance(properties, dict):
            spec = properties.get("KeySpec")
            if spec and isinstance(spec, str):
                if 'SYMMETRIC_DEFAULT' not in spec and 'HMAC' not in spec:
                    return CheckResult.UNKNOWN
        return super().scan_resource_conf(conf)
```
the result should be `UNKNOWN`, because as you said, you can't enable key rotation for asymmetric keys, so the check is actually not valid for this setup. And I also suggest a bit different logic to be more on the safer side 🙂